addons/mail/models/discuss/ir_websocket.py

```diff
@@ -22,10 +24,16 @@ def _get_im_status(self, data):
     @add_guest_to_context
     def _build_bus_channel_list(self, channels):
         channels = list(channels)  # do not alter original list
+        channel_ids = list()
+        for channel in channels:
+            if isinstance(channel, str):
+                match = re.findall(r'discuss\.channel.*?(\b\d+\b)', channel)
+                if match:
+                    channel_ids.append(int(match[0]))
```

Comment:
Ideally we should pop matching str from `channels` as we don't want to actually subscribe to those str channels in the parent.